devito/finite_differences/differentiable.py

```diff
@@ -288,6 +288,15 @@ def subs(self, *args, **kwargs):
         return self.func(*[getattr(a, 'subs', lambda x: a)(*args, **kwargs)
                            for a in self.args], evaluate=False)
 
+    def _subs(self, old, new, **hints):
```

Comment:
out of curiosity, how is this different than the one in sympy?

Reply:
so, I just checked sympy.Basic._subs

so now this overriding does something different, so we should probably at least update the docstring

as I wrote in another comment below, I'm still a bit baffled by the lack of a check for identity (ie something along the lines as `if self is old then ...`. 

Also check this out: https://github.com/sympy/sympy/blob/master/sympy/core/basic.py#L998